waiter/src/waiter/handler.clj

```diff
@@ -1282,3 +1282,96 @@
+                ;;let [instance (service/get-rand-inst populate-maintainer-chan! service-id (reason-map-fn) exclude-ids-set inter-kill-request-wait-time-ms)]
+                (let [service-id (get-service-id-from-instance-id instance-id)]
+                    (def instance {:id instance-id :service-id service-id})
+                      (do
+                        ;; CHANGED INSTANCE TO INSTANCE-ID
+                        (log/info "sending sigkill to instance " instance-id service-id)
+                        ;; (counters/inc! (metrics/service-counter service-id "scaling" "scale-down" "attempt"))
+                        (scheduler/track-kill-candidate! instance-id :prepare-to-kill eject-backoff-base-time-ms)
+                        (let [{:keys [killed?] :as kill-result}
+                              (-> (au/execute
+                                    (fn send-signal-to-instance []
+                                      (scheduler/signal-instance scheduler instance :sigkill))
+                                    thread-pool)
+                                  async/<!
+                                  :result)]
+                          (if killed?
+                            (do
+                              (log/info "marking instance" instance-id "as killed")
+                              (counters/inc! (metrics/service-counter service-id "instance-counts" "killed"))
+                              ;; (counters/inc! (metrics/service-counter service-id "scaling" "scale-down" "success"))
+                              (scheduler/track-kill-candidate! instance-id :killed eject-backoff-base-time-ms)
+                              ;; (service/release-instance! populate-maintainer-chan! instance (result-map-fn :killed))
+                              (notify-instance-killed-fn instance))
+                            (do
+                              (log/info "failed kill attempt, releasing instance" instance-id)
+                              ;; (counters/inc! (metrics/service-counter service-id "scaling" "scale-down" "kill-fail"))
+                              (service/release-instance! populate-maintainer-chan! instance (result-map-fn :not-killed))))
+                          (when response-chan (async/>! response-chan kill-result))
+                          killed?))
+                    ;; (do
+                      ;; (log/info "no instance available to kill")
+                      ;; (counters/inc! (metrics/service-counter service-id "scaling" "scale-down" "unavailable"))
+                      ;; false)
+                               ))
+          (catch Exception ex
+            ;; (counters/inc! (metrics/service-counter service-id "scaling" "scale-down" "fail"))
+            (log/error ex "unable to send signal to instance " instance-id)
+            (when response-chan (async/>! response-chan {:instance-id instance-id :status http-500-internal-server-error :message (.getMessage ex)}))))))))
+
+
+
+(defn signal-handler
```

Comment:
- Should only accept POST requests
- Need authorization checks with `allowed-to-manage-service?-fn`